### PR TITLE
GcnEventPage async

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR adds both async queries to the GcnEventPage and indexes the galaxy and observation table in an attempt to improve GcnEventPage performance.

### CODE REVIEW

**Comment by Theodlz** [comment]
Hi @mcoughlin, as discussed on slack, as some things are debatable and clearly need to be improved on the GcnEventPag, I created a branch on your fork with suggestions/ideas for how to do some things differently. It is this one: https://github.com/mcoughlin/skyportal/tree/obs_async_theo

File: static/js/components/GcnEventPage.jsx [hunk 1]
```diff
@@ -251,23 +251,39 @@ const GcnEventPage = ({ route }) => {
...
+      await dispatch(gcnEventActions.fetchGcnEvent(dateobs));
+    };
+    fetchGcnEvent(route.dateobs);
   }, [route, dispatch]);
 
   // useEffect(() => {
+  // const fetchGcnEventSources = async (dateobs) => {
+  //    await dispatch(sourcesActions.fetchGcnEventSources(dateobs));
+  // }
+  //  fetchGcnEventSources(route.dateobs);
```
Comment (guynir42) [inline static/js/components/GcnEventPage.jsx:264]: can we get rid of the commented code?

File: static/js/components/GcnSelectionForm.jsx [hunk 2]
```diff
@@ -17,7 +17,7 @@ import utc from "dayjs/plugin/utc";
 import relativeTime from "dayjs/plugin/relativeTime";
 
 import { filterOutEmptyValues } from "../API";
-import * as sourcesActions from "../ducks/sources";
+// import * as sourcesActions from "../ducks/sources";
```
Comment (guynir42) [inline static/js/components/GcnSelectionForm.jsx:20]: More commented code here and below.

**Review by guynir42 — APPROVED** [review]
Minor comments below. Looks good. Please make sure the migration on this and the other PRs do not collide...

**Comment by mcoughlin** [comment]
@guynir42 i wanna keep them as I will be uncommenting when your source PR goes in ;)

**Comment by guynir42** [comment]
That's fine. Maybe add a comment explaining that this will be uncommented when "last detected" is finished.

File: static/js/components/GcnSelectionForm.jsx [hunk 3]
```diff
@@ -206,7 +207,7 @@ const GcnSelectionForm = ({ gcnEvent, setSelectedLocalizationName }) => {
 
   if (
     !gcnEvent ||
-    // !gcnEventSources ||
+    !gcnEventSources ||
```
Comment (Theodlz) [inline static/js/components/GcnSelectionForm.jsx:210]: Couldn't we still charge the gcn selection form as soon as we have the event and its localization ? And then the options to display or not display sources, observations, instruments, galaxies ... appear as those become available? Or they are already there the buttons are deactivated and they get clickable as it loads ?

Comment (Theodlz) [reply static/js/components/GcnSelectionForm.jsx:210]: Just to follow the async behaviour we expect from the backend you know. And to not end up having a whole set of features deactivated when just one of them really is.

Comment (mcoughlin) [reply static/js/components/GcnSelectionForm.jsx]: @Theodlz I pushed a commit with your disable button trick... give it a try when you have a chance.

**Review by Theodlz — APPROVED** [review]
Much better !

**Pull request merged**